File: 092/092.c
#include <stdio.h>

int squareDigits(int n) {
    int res = 0;
    while (n > 0) {
        int d = n % 10;
        res += d * d;
        n /= 10;
    }
    return res;
}

int main() {
    int count = 0;
    for (int i = 2; i < 1e7; i++) {
        int x = i;
        while (x != 1) {
            if (x == 89) {
                count++;
                break;
            }
            x = squareDigits(x);
        }
    }
    printf("%d\n", count);
}

File: solutions/092/092.go
package main

import (
	"fmt"
)

func squareDigits(n int) (res int) {
	for n > 0 {
		d := n % 10
		n /= 10
		res += d * d
	}
	return
}

func main() {
	count := 0
	for i := 2; i < 10000000; i++ {
		x := i
		for x != 1 {
			if x == 89 {
				count++
				break
			}
			x = squareDigits(x)
		}
	}
	fmt.Println(count)
}
